tests/test_hdf_ioc.py

```diff
@@ -384,6 +391,53 @@ async def test_hdf5_ioc_parameter_validate_works(
     assert val == str(tmp_path) + "/name.h5"  # put should have been stopped
 
 
+@pytest.mark.parametrize(
+    "create_depth, path, expect_exists",
+    [
+        (0, "panda_test1", False),
+        (-2, "panda_test2", True),
+        (-1, "panda_test3/depth_2", False),
+        (1, "panda_test4/depth_2", True),
+        (0, "/", False),
+        (-1, "/panda_test5", False),
+        # make sure our final test passes, so final status message is "OK"
+        (0, ".", True),
+    ],
+)
+async def test_hdf5_directory_creation(
+    hdf5_subprocess_ioc,
+    tmp_path: Path,
+    create_depth: int,
+    path: str,
+    expect_exists: bool,
+):
+    """Test to see if directory creation/exists works as expected"""
+
+    _, hdf5_test_prefix = hdf5_subprocess_ioc
+
+    target_path = str(os.path.join(tmp_path, path))
+    if path.startswith("/"):
+        target_path = path
```

Comment:
I think this runs the risk of potentially creating the `/panda_test5` folder in someone's root directory, if permissions are not as we expect?

Reply:
Yes, I guess so. Is there a directory I can use as one that is expected to exist but not be owned by the user running the tests?

Reply:
I ended up just changing the permissions for the `tmp_path` dir at the start of the test to something not writable, then changing them back after.